Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py

```diff
@@ -0,0 +1,919 @@
+        )
+
+    def create_local_ca(self, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=LOCAL_CAS_URL_SUFFIX,
+            json_data=request_data,
+        )
+
+    def get_local_ca_list(self, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='GET',
+            url_suffix=LOCAL_CAS_URL_SUFFIX,
+            params=params,
+        )
+
+    def get_local_ca(self, local_ca_id: str, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='GET',
+            url_suffix=urljoin(LOCAL_CAS_URL_SUFFIX, local_ca_id),
+            params=params,
+        )
+
+    def update_local_ca(self, local_ca_id: str, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='PATCH',
+            url_suffix=urljoin(LOCAL_CAS_URL_SUFFIX, local_ca_id),
+            params=params,
+        )
+
+    def delete_local_ca(self, local_ca_id: str) -> dict[str, Any]:
+        return self._http_request(
+            method='DELETE',
+            url_suffix=urljoin(LOCAL_CAS_URL_SUFFIX, local_ca_id),
+            return_empty_response=True,
+        )
+
+    def self_sign_local_ca(self, local_ca_id: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, local_ca_id)}/self-sign',
+            json_data=request_data,
+        )
+
+    def install_local_ca(self, local_ca_id: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, local_ca_id)}/install',
+            json_data=request_data,
+        )
+
+    def issue_certificate(self, ca_id: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, ca_id)}/certs',
+            json_data=request_data,
+        )
+
+    def get_certificates_list(self, ca_id: str, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='GET',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, ca_id)}/certs',
+            params=params,
+        )
+
+    def delete_certificate(self, ca_id: str, local_ca_id: str) -> dict[str, Any]:
+        return self._http_request(
+            method='DELETE',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, ca_id)}/certs/{local_ca_id}',
+            return_empty_response=True,
+        )
+
+    def revoke_certificate(self, ca_id: str, cert_id: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=f'{urljoin(LOCAL_CAS_URL_SUFFIX, ca_id)}/certs/{cert_id}/revoke',
+            json_data=request_data,
+            return_empty_response=True,
+            empty_valid_codes=[200],
```

Comment:
> raised exception otherwise since the api returns an empty response. see _handle_sucsses method that is called by the _http_request: ![image](https://private-user-images.githubusercontent.com/94686128/336534324-df523b14-f8e7-46be-b16d-66b273847673.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1NzY3NTMsIm5iZiI6MTcxNzU3NjQ1MywicGF0aCI6Ii85NDY4NjEyOC8zMzY1MzQzMjQtZGY1MjNiMTQtZjhlNy00NmJlLWIxNmQtNjZiMjczODQ3NjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDA4MzQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3MzhjNjI2ZjU1MzQxNjQxZGYxNDk1NzEyN2EyMTZhNzc1MGFjYmIxZmExYzUxNDU2MGE4NGZmM2YzMTkxNWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rILfU31WVOtdNNMWrcrtKZ9Txs5OtZgDQHufGf_Z0Gc) ![image](https://private-user-images.githubusercontent.com/94686128/336534354-aecb9b82-2128-4581-a172-54d23dce4e25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1NzY3NTMsIm5iZiI6MTcxNzU3NjQ1MywicGF0aCI6Ii85NDY4NjEyOC8zMzY1MzQzNTQtYWVjYjliODItMjEyOC00NTgxLWExNzItNTRkMjNkY2U0ZTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDA4MzQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1NTJmMzJhNzZkNWRkODA4NTc4ZmQxMmFkYTc1YjZhOGI0ZmFhNDE1MTkyYWJjYzEzMjJkZTlhMGY3NTJiMmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oLVa2HFzg4L8uqoOabsp6lntc41hxK32_z6w8YaC1Fo) ![image](https://private-user-images.githubusercontent.com/94686128/336534380-bfb6c922-f0a2-46fc-a320-0641d389b165.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1NzY3NTMsIm5iZiI6MTcxNzU3NjQ1MywicGF0aCI6Ii85NDY4NjEyOC8zMzY1MzQzODAtYmZiNmM5MjItZjBhMi00NmZjLWEzMjAtMDY0MWQzODliMTY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDA4MzQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4NTFiMTY5ZTMyOTJkMzZjNTk4ZGM4YmRmYWE4NDcyZDAwNDc1ZmQ4NWFjNDc2Y2EwNTMzMzg0YjczNzAxNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XyJCGOLT5jDITdSA4ZASKRYgtr1wCPWX-RCsLbtGi9g)

Reply:
My mistake, when walking initially through their api documentation it seemed that the response is empty: 
![image](https://github.com/demisto/content/assets/94686128/8a17dd57-32e2-494b-a4a0-85f59fa36cfd)